lib/Runtime/Types/RecyclableObject.cpp

```diff
@@ -197,6 +197,18 @@ namespace Js
         return true;
     }
 
+    bool RecyclableObject::HasAnySpecialProperties()
+    {
+        if (DynamicType::Is(this->GetTypeId()))
```

Comment:
I recently ran into an issue in another change where CustomExternalObject could override behavior in creepy ways, so it acts like it has a property even when the underlying dynamic type doesn't know about those properties. Do you anticipate that we could see similar problems here?

Reply:
I believe this is handled by change to ExternalType in ChakraFull (now linked below).

---
In reply to: [181897202](https://github.com/Microsoft/ChakraCore/pull/4993#discussion_r181897202) [](ancestors = 181897202)

Reply:
Great, thanks!

---
In reply to: [181918396](https://github.com/Microsoft/ChakraCore/pull/4993#discussion_r181918396) [](ancestors = 181918396,181897202)